### ISSUE TITLE
Hello, when I run Mavsdk-Swift-Example, after connecting the UAV, the CPU will reach 30%. What causes this problem? Can it be solved?

### ISSUE THREAD

**Issue by fanfans1** [mavlink/MAVSDK-Swift#220]

**Comment by fanfans1** [comment]
Running Mavsdk-Swift-Example at iOS

**Comment by fanfans1** [comment]
Running Mavsdk-Swift-Example at iOS

**Comment by JonasVautherin** [comment]
Difficult to know without knowing which device you are using, which version of MAVSDK, etc.

**Comment by fanfans1** [comment]
MAVSDK version: v1.4.2, iPhone X。

**Comment by JonasVautherin** [comment]
Interesting. I never profiled it on an iPhone, I don't know what to expect there. Does it consistently stay at 30% during the flight?

**Comment by fanfans1** [comment]
Yes, our drone temporarily runs on simulators, sometimes the CPU reaches 50




---Original---
From: "Jonas ***@***.***&gt;
Date: Mon, Nov 7, 2022 18:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mavlink/MAVSDK-Swift] Hello, when I run Mavsdk-Swift-Example, after connecting the UAV, the CPU will reach 30%. What causes this problem? Can it be solved? (Issue #220)




 
Interesting. I never profiled it on an iPhone, I don't know what to expect there. Does it consistently stay at 30% during the flight?

**Comment by JonasVautherin** [comment]
Without the video stream?

**Comment by fanfans1** [comment]
No, it contains video streams



---Original---
From: "Jonas ***@***.***&gt;
Date: Mon, Nov 7, 2022 20:29 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mavlink/MAVSDK-Swift] Hello, when I run Mavsdk-Swift-Example,after connecting the UAV, the CPU will reach 30%. What causes this problem?Can it be solved? (Issue #220)




 
Without the video stream?

**Comment by JonasVautherin** [comment]
Can you try without the video streams, for comparison?

**Comment by fanfans1** [comment]
I tried, without video streaming CUP is about 2% lower, after changing to the iPhone 11, CUP becomes about 18%



---Original---
From: "Jonas ***@***.***&gt;
Date: Mon, Nov 7, 2022 22:16 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mavlink/MAVSDK-Swift] Hello, when I run Mavsdk-Swift-Example, after connecting the UAV, the CPU will reach 30%. What causes this problem? Can it be solved? (Issue #220)




 
Can you try without the video streams, for comparison?

**Comment by julianoes** [comment]
Sounds like video streaming is the reason then? Make sure to use video encoding in hardware. Not sure how that works on iOS but I'm sure there are guides.
